scripts/generate-cli-completions.py

```
@@ -0,0 +1,419 @@
+            lines.append(f"                COMPREPLY=( $(compgen -W \"{' '.join(vals)}\" -- \"$cur\") ); return")
+            lines.append("            fi")
+        lines.append(f"            COMPREPLY=( $(compgen -W \"{' '.join(words)}\" -- \"$cur\") ); return ;;")
+    lines.append("    esac")
+    lines.append("}")
+    lines.append("complete -F _cmux cmux")
+    lines.append("")
+    return "\n".join(lines)
+
+
+def emit_zsh(commands: list[str], specs: dict[str, CommandSpec]) -> str:
+    lines = ["#compdef cmux", HEADER.format(shell="zsh"), "_cmux() {"]
```

Ruff (0.15.18)

[error] Function call with truthy `shell` parameter identified, security issue

(S604)